tests/components/myuplink/test_config_flow.py

```diff
@@ -81,3 +81,60 @@ async def test_full_flow(
 
     assert len(hass.config_entries.async_entries(DOMAIN)) == 1
     assert len(mock_setup.mock_calls) == 1
+
+
+async def test_flow_reauth(
+    hass: HomeAssistant,
+    hass_client_no_auth,
+    aioclient_mock,
+    current_request_with_host,
+    setup_credentials,
+    init_integration,
+    expires_at,
+) -> None:
+    """Test reauth step."""
+
+    OLD_SCOPE = {
+        "auth_implementation": DOMAIN,
+        "token": {
+            "access_token": "Fake_token",
+            "scope": "READSYSTEM offline_access",
+            "expires_in": 86399,
+            "refresh_token": "3012bc9f-7a65-4240-b817-9154ffdcc30f",
+            "token_type": "Bearer",
+            "expires_at": expires_at,
+        },
+    }
+
+    entry = init_integration
+    hass.config_entries.async_update_entry(entry, data=OLD_SCOPE)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": entry.entry_id,
+        },
+        data=entry.data,
+    )
+
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+    assert "flow_id" in flows[0]
+    assert result["flow_id"] == flows[0]["flow_id"]
```

Comment:
We don't need to check current flows. We already have a handle to the flow via the result flow id.

Please assert the step id of each result. It's much easier for everyone including yourself to see where the flow is at.